### DISCUSSION TITLE
CLI: "user mod" is broken: "username '<user>' already exists"

### DISCUSSION THREAD

**Discussion by x86dev** [photoprism/photoprism#3135] [Q&A]
When trying to modify an existing users, (m *User) SetValuesFromCli() calls m.Validate() at the end, which in turn checks if the user in the DB already exists and therefore bails out.

**Comment by graciousgrey** [comment]
Thank you very much for your report.

We receive a large number of requests every day, so it may take a while before we can get back to you. Thank you for your patience.

Since many "potential bugs" are caused by individual tech stacks/configuration, please go through our [troubleshooting guides](https://docs.photoprism.app/getting-started/troubleshooting/) and close this issue if it could be solved by reading the documentation.

Please note that we are only able to process bug reports that contain all relevant information listed in our [issue template](https://github.com/photoprism/photoprism/blob/develop/.github/ISSUE_TEMPLATE/bug_report.md).

Incomplete reports will be marked as such and closed after a grace period if no additional information is provided.

**Comment by lastzero** [comment]
Works for me... might this be your issue? https://docs.photoprism.app/known-issues/#user-authentication

```
photoprism@230111-jammy:/go/src/github.com/photoprism/photoprism$ ./photoprism users ls
INFO[2023-01-20T11:23:30Z] found 1 user                                 

|------------------|-------|-----------|-------|-------|-------------|---------|---------|------------|---------------------|
|       User       | Login | Full Name | Email | Role  | Super Admin | Web UI  | WebDAV  | Attributes |     Created At      |
|------------------|-------|-----------|-------|-------|-------------|---------|---------|------------|---------------------|
| urobw8s1knzbb3sr | admin | Admin     |       | admin | yes         | enabled | enabled |            | 2023-01-11 15:55:40 |
|------------------|-------|-----------|-------|-------|-------------|---------|---------|------------|---------------------|

photoprism@230111-jammy:/go/src/github.com/photoprism/photoprism$ ./photoprism users mod -n Foobar admin     
INFO[2023-01-20T11:23:52Z] user admin has been updated                  
```

**Comment by x86dev** [comment]
I'm running "230117-41774fcec-Linux-AMD64":

I just tried it out again, now it seems to work ... strange. Is there any way via CLI to enable "Web UI" login for users with the "visitor" role yet?

```
photoprism users mod --no-login=False myuser
```

doesn't do anything.

**Reply by lastzero** [reply]
Use link sharing. That's the visitor role. Also, false should be lowercase.

**Reply by x86dev** [reply]
Thanks for your fast response and shedding some light on this!
